p2p: new non-deterministic bug in test

From a PR with no code changes ( https://github.com/tendermint/tendermint/pull/2149 - https://circleci.com/gh/tendermint/tendermint/18680#tests/containers/2):

```
ok  	github.com/tendermint/tendermint/node	1.124s	coverage: 70.8% of statements
panic: runtime error: index out of range

goroutine 271 [running]:
github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).dialSeeds(0xc4203bc9c0)
	/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:516 +0x4df
github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).ensurePeers(0xc4203bc9c0)
	/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:428 +0xbcb
github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).ensurePeersRoutine(0xc4203bc9c0)
	/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:342 +0xf0
created by github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).OnStart
	/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:138 +0x27f
FAIL	github.com/tendermint/tendermint/p2p/pex	0.733s
Exited with code 1
```

--- a/p2p/pex/pex_reactor.go
+++ b/p2p/pex/pex_reactor.go
@@ -504,12 +504,7 @@ func (r *PEXReactor) checkSeeds() (numOnline int, netAddrs []*p2p.NetAddress, er
 
 // randomly dial seeds until we connect to one or exhaust them
 func (r *PEXReactor) dialSeeds() {
-	lSeeds := len(r.config.Seeds)
-	if lSeeds == 0 {
-		return
-	}
-
-	perm := cmn.RandPerm(lSeeds)
+	perm := cmn.RandPerm(len(r.seedAddrs))
 	// perm := r.Switch.rng.Perm(lSeeds)
 	for _, i := range perm {
 		// dial a random seed

